src/store/modules/job/actions.ts

```diff
@@ -762,6 +763,137 @@ const actions: ActionTree<JobState, RootState> = {
+    }
+    payload.runtime = state.bulk.runtime;
+    // set the maximum slow frequency for slow type jobs if global frequnecy is set
+    payload.frequency = (state.bulk.frequency && freqType === 'slow') ? (generateFrequencyOptions('slow') as any).pop().value : state.bulk.frequency;
+    
+    const bulk = JSON.parse(JSON.stringify(state.bulk.jobs));
+    bulk.push(payload)
+    commit(types.JOB_BULK_UPDATED, bulk);
+  },
+  async scheduleBulkJobs({ dispatch }, payload) {
+    // TODO: handle case for bulk jobs which are already scheduled. 
+    // As of now, the job will get scheduled again even if it is pending
+    const jobParams = [] as any;
+    let failedJobs = 0;
+    payload.shopifyConfigs.map((shopId: string) => {
+      return payload.jobs.reduce((params: any, job: any) => {
+        params = {
+          'JOB_NAME': job.jobName,
+          'SERVICE_NAME': job.serviceName,
+          'SERVICE_COUNT': '0',
+          'SERVICE_TIME': job.runtime.toString(),
+          'SERVICE_TEMP_EXPR': job.frequency,
+          'SERVICE_RUN_AS_SYSTEM': 'Y',
+          'jobFields': {
+            'productStoreId': payload.eComStoreId,
+            'systemJobEnumId': job.systemJobEnumId,
+            'tempExprId': job.frequency, // Need to remove this as we are passing frequency in SERVICE_TEMP_EXPR, currently kept it for backward compatibility
+            'maxRecurrenceCount': '-1',
+            'parentJobId': job.parentJobId,
+            'runAsUser': 'system', //default system, but empty in run now.  TODO Need to remove this as we are using SERVICE_RUN_AS_SYSTEM, currently kept it for backward compatibility
+            'recurrenceTimeZone': this.state.user.current.userTimeZone
+          },
+          'statusId': "SERVICE_PENDING",
+          'systemJobEnumId': job.systemJobEnumId
+        } as any
+
+        
+        if (job?.runtimeData?.shopifyConfigId) {
+          const shopifyConfig = this.state.user.shopifyConfigs.find((config: any) => {
+            return config.shopId === shopId;
+          })
+
+          params['shopifyConfigId'] = shopifyConfig.shopifyConfigId;
+          params['jobFields']['shopId'] = shopId;
+        }
+
+        // checking if the runtimeData has productStoreId, and if present then adding it on root level
+        job?.runtimeData?.productStoreId?.length >= 0 && (params['productStoreId'] = payload.eComStoreId)
+        job?.priority && (params['SERVICE_PRIORITY'] = job.priority.toString())
+
+        // assigning '' (empty string) to all the runtimeData properties whose value is "null"
+        job.runtimeData && Object.keys(job.runtimeData).map((key: any) => {
+          if (job.runtimeData[key] === 'null') job.runtimeData[key] = ''
+        })
+        jobParams.push({ ...job.runtimeData, ...params });
+      }, {})
+    })
+
+    Promise.allSettled(jobParams.map(async (param: any) => {
```

Comment:
We could use async await

Reply:
Sir, the use of Promise.all will reject the next calls if a previous one fails.